cumulusci/salesforce_api/utils.py

```diff
@@ -12,12 +13,27 @@
 def get_simple_salesforce_connection(
     project_config, org_config, api_version=None, base_url: str = None
 ):
+
     # Retry on long-running metadeploy jobs
     retries = Retry(total=5, status_forcelist=(502, 503, 504), backoff_factor=0.3)
     adapter = HTTPAdapter(max_retries=retries)
+    instance = org_config.instance_url
+
+    # Attept to get the host and port from the URL, but ignore any errors retrieving it.
+    try:
```

Comment:
Just a tip to help you level-up in Python.

try/except: pass will not survive a code review and most linters will also flag it. It will swallow all sorts of weird stuff like Ctrl-C signals, out-of-memory signals etc.

```
try:

except Exception:
   pass
```

is very slightly better, but not as good as good as catching just the exception that you see being raised.

But I'll take a look at what changes to the test suite might make this unnecessary altogether.

Reply:
Thanks. I had to add the try/except blocks because without them the tests (almost all) were failing with the following error. I think it's because the mocked URL that gets passed to this method (in tests) is an invalid URL that causes the urlparse library to throw an exception.
`TypeError: '>' not supported between instances of 'Mock' and 'int'`